src/main/webapp/app/overview/course-faq/course-faq.component.ts

```diff
@@ -51,17 +55,31 @@ export class CourseFaqComponent implements OnInit, OnDestroy {
 
     private faqService = inject(FaqService);
     private alertService = inject(AlertService);
+    private sortService = inject(SortService);
+    private renderer = inject(Renderer2);
 
     ngOnInit(): void {
         this.parentParamSubscription = this.route.parent!.params.subscribe((params) => {
             this.courseId = Number(params.courseId);
             this.loadFaqs();
             this.loadCourseExerciseCategories(this.courseId);
         });
+
+        this.route.queryParams.subscribe((params) => {
```

Comment:
In the course-faq-components.spec.ts, the tests seem to be failing because `queryParams` is undefined. I think you'll have to update the test setup. Can you have a look?